### ISSUE TITLE
Splashscreen iOS implementation

### ISSUE THREAD

**Issue by export-mike** [crazycodeboy/react-native-splash-screen#463]
Hi this code for ios seems like it needs some review.
https://github.com/crazycodeboy/react-native-splash-screen/blob/a6ab78b70e81c8e003f8f109b19c7324ebb2e487/ios/SplashScreen.m#L30

Its preventing apps from using e2e testing libraries. 

see detox issue: https://github.com/wix/Detox/issues/400

**Comment by export-mike** [comment]
I ended up adding a release mode flag to the app when it’s built for e2e testing preventing the splashcreen from showing

**Comment by mbrehme** [comment]
@export-mike would you mind showing us how you did this. It would help alot!

Thanks in advanced

**Comment by Abramovick** [comment]
@export-mike Also facing the same issue. The test used to work fine, but now they just hang on the SplashScreen and timeout without anything working at all. 
Any tips on how you fixed it (if you did) would be great.

Ta.

**Comment by halilb** [comment]
@mbrehme @Abramovick 

I'm using this block to initialize SplashScreen to fix the issue:

```
bool isDetox = [[[NSProcessInfo processInfo] arguments] containsObject: @"-detoxServer"];
if (!isDetox) {
  [RNSplashScreen show];
}
```

**Comment by export-mike** [comment]
yeah I was doing something similar but with a combination of react-native-config

**Comment by export-mike** [comment]
@Abramovick @mbrehme 

> yeah I was doing something similar but with a combination of react-native-config

**Comment by dazlious** [comment]
this is still an ongoing issue
